src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -190,7 +190,7 @@ public Collection<Range<Token>> getPrimaryRangesWithinDC(String keyspace)
     private double traceProbability = 0.0;
 
     @VisibleForTesting
-    static enum Mode { STARTING, NORMAL, JOINING, LEAVING, DECOMMISSIONED, MOVING, DRAINING, DRAINED, ZOMBIE, NON_TRANSIENT_ERROR, TRANSIENT_ERROR, WAITING_TO_BOOTSTRAP, DISABLED }
+    static enum Mode { STARTING, NORMAL, JOINING, LEAVING, DECOMMISSIONED, MOVING, DRAINING, DRAINED, ZOMBIE, NON_TRANSIENT_ERROR, TRANSIENT_ERROR, WAITING_TO_BOOTSTRAP, WAITING_TO_FINISH_BOOTSTRAP, DISABLED }
```

Comment:
we should include this in all the places that have logic based on `JOINING`

also, can you validate that writes will continue to be forwarded to this node in the new operation mode?

Reply:
confirmed that write behavior is not related to OperationMode

Reply:
Also confirmed that all other `JONING` related logic is only for logging.